operators/src/processing/expression/query_processor.rs

```diff
@@ -389,46 +399,52 @@ macro_rules! impl_expression_tuple_processor {
                 program: &LinkedExpression,
                 map_no_data: bool,
                 out_no_data: TO,
-            ) -> Result<Vec<TO>> {
+            ) -> Result<GridOrEmpty2D<TO>> {
                 let expression: Symbol<$FN_T> = unsafe {
                     // we have to "trust" that the function has the signature we expect
                     program.function_nary()?
                 };
 
-                let min_batch_size = rasters[0].grid_array.grid_shape().axis_size_x();
+                let shape = rasters[0].grid_shape().clone();
+
+                let mut out = GridOrEmpty::from(EmptyGrid2D::<TO>::new(shape));
 
-                let [ $($TILE),* ] = rasters.map(|raster| raster.into_materialized_tile());
+                let map_fn = move |lin_idx: usize, _empty: Option<TO>| {
+                    $(
+                        let $PIXEL = rasters[$I].get_at_grid_index_unchecked(lin_idx);
```

Comment:
why can you just use a lin_idx as a gridIndex here? Aren't grid index n-dimensional?